sdks/python/apache_beam/transforms/external.py

```diff
@@ -42,6 +44,63 @@
   grpc = None
 # pylint: enable=wrong-import-order, wrong-import-position, ungrouped-imports
 
+DEFAULT_EXPANSION_SERVICE = 'localhost:8097'
+FILTERED_CODERS = {
+  'beam:coder:length_prefix:v1',
+}
+
+
+def iter_urns(coder, context=None):
+  yield coder.to_runner_api_parameter(context)[0]
+  for child in coder._get_component_coders():
+    for urn in iter_urns(child, context):
+      yield urn
+
+
+class External(ptransform.PTransform):
+  """
+  Base class for transforms that produce external transforms.
+  """
+  _urn = None
+
+  def __init__(self, expansion_service=None):
+    self.expansion_service = expansion_service or DEFAULT_EXPANSION_SERVICE
+
+  def get_config_args(self):
+    """
+    :return: dictionary of encoded configuration values to pass to
+             ExternalConfigurationPayload
+    """
+    raise NotImplementedError
+
+  @classmethod
+  def config_value(cls, obj, coder):
+    """
+    Helper to create a ConfigValue with an encoded value.
+    """
+    return ConfigValue(
+      coder_urn=[urn for urn in iter_urns(coder)
```

Comment:
@mxm So is this just a flattened list of all components? What about coders with payloads? Wasn't the plan instead to reference coder ids from the outer pipeline components?

Reply:
Or was this just a stopgap until we had schemas anyway? (Probably not worth tackling in this PR.)

Reply:
Once the portable schema stuff is wrapped up, hopefully we can migrate to that as a more standard format.